upstairs/src/lib.rs

```diff
@@ -1617,15 +1630,15 @@ impl Buffer {
         assert_eq!(offset % self.block_size, 0);
         assert_eq!(response.data.len(), self.block_size);
         if !response.block_contexts.is_empty() {
-            self.data[offset..][..response.data.len()]
+            self.block_mut(offset / self.block_size)
                 .copy_from_slice(&response.data);
-            self.owned[offset..][..response.data.len()].fill(true);
+            self.owned[offset / self.block_size] = true;
         }
     }
 
     /// Reads buffer data into the given array
     ///
-    /// Values without `self.owned` are left unmodified
+    /// Values in blocks with `self.owned` are left unmodified
```

Comment:
I think this inverted the meaning? Maybe:

> Only values in blocks with `self.owned` are modified

Reply:
Good catch, done in 8769d72db6de05b4179c5196b388ae3d7b142069

Reply:
:+1: